spring-kafka/src/main/java/org/springframework/kafka/listener/DeadLetterPublishingRecoverer.java

```diff
@@ -195,6 +232,30 @@ public void accept(ConsumerRecord<?, ?> record, Exception exception) {
 		}
 	}
 
+	private TopicPartition checkPartition(TopicPartition tp, Consumer<?, ?> consumer) {
+		if (tp.partition() < 0) {
+			return tp;
+		}
+		try {
+			List<PartitionInfo> partitions = consumer.partitionsFor(tp.topic(), this.partitionInfoTimeout);
+			if (partitions == null) {
+				this.logger.debug(() -> "Could not obtain partition info for " + tp.topic());
+				return tp;
+			}
+			boolean anyMatch = partitions.stream().anyMatch(pi -> pi.partition() == tp.partition());
+			if (!anyMatch) {
+				this.logger.warn(() -> "Destination resolver returned non-existent partition " + tp
+						+ ", KafkaProducer will determine partition to use for this topic");
+				return new TopicPartition(tp.topic(), -1);
+			}
+			return tp;
+		}
+		catch (Exception ex) {
+			this.logger.debug(ex, () -> "Could not obtain partition info for " + tp.topic());
```

Comment:
Just to double check: is it intentional to have it as `debug`  over here on this error and it is not a copy/paste artifact from the code above?
I mean returned wrong partition is a `warn`, but this exception is `debug` 🤷

Reply:
It's intentional - we might not have permissions and I didn't want to flood the logs.